x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateMetricDoubleFieldMapper.java

```diff
@@ -215,12 +217,22 @@ public AggregateMetricDoubleFieldMapper build(MapperBuilderContext context) {
                 }
             }
 
-            if (metrics.getValue().contains(defaultMetric.getValue()) == false) {
+            if (metrics.getValue().contains(defaultMetric.getValue()) == false && defaultMetric.getValue() != Metric.avg) {
                 // The default_metric is not defined in the "metrics" field
                 throw new IllegalArgumentException(
                     "Default metric [" + defaultMetric.getValue() + "] is not defined in the metrics of field [" + leafName() + "]."
                 );
             }
+            if (defaultMetric.getValue() == Metric.avg
+                && (metrics.getValue().contains(Metric.sum) == false && metrics.getValue().contains(Metric.value_count) == false)) {
```

Comment:
If I am not mistaken we need an or here, because if any of them is not available we cannot calculate the average.
```suggestion
                && (metrics.getValue().contains(Metric.sum) == false || metrics.getValue().contains(Metric.value_count) == false)) {
```